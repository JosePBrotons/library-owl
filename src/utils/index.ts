export const isSafeDataType = (data: any): boolean => {
    switch (Object.prototype.toString.call(data)) {
        case '[object Null]':
            return false;
        case '[object Undefined]':
            return false;
        default:
            return true;
    }
};

export const isString = (value: any) =>
    Object.prototype.toString.call(value) === '[object String]';

export const isBlank = (value: string = '') => {
    if (isString(value)) {
        const delEmptySpaces = value.replace(/\s/g, '');
        return delEmptySpaces.length === 0;
    }
    return false;
};
export const firstString = (value: string = '', quantity: number) => {
    return isString(value) ? value.substring(0, quantity) : '';
};
export const lastString = (value: string, quantity: number) => {
    return isString(value)
        ? value.substring(value.length, value.length - quantity)
        : '';
};

export const isArrayLength = (
    arr: Array<any> = [],
    condition: string = 'equal',
    value: number = 0
): boolean => {
    if (Array.isArray(arr)) {
        const arrCond = {
            greater: (arrToCheck: Array<any>, arrValue: number) =>
                arrToCheck.length > arrValue,
            lower: (arrToCheck: Array<any>, arrValue: number) =>
                arrToCheck.length < arrValue,
            equal: (arrToCheck: Array<any>, arrValue: number) =>
                arrToCheck.length === arrValue,
            lowOrEq: (arrToCheck: Array<any>, arrValue: number) =>
                arrToCheck.length <= arrValue,
            greatOrEq: (arrToCheck: Array<any>, arrValue: number) =>
                arrToCheck.length >= arrValue,
        };
        return !!arrCond[condition as keyof typeof arrCond]
            ? arrCond[condition as keyof typeof arrCond](arr, value)
            : false;
    }
    return false;
};

export const isValueLength = (
    value: number,
    condition: string = 'equal',
    valueCompare: number = 0
): boolean => {
    const conditions = {
        greater: (valueToCheck: number, _value: number) =>
            valueToCheck > _value,
        lower: (valueToCheck: number, _value: number) => valueToCheck < _value,
        equal: (valueToCheck: number, _value: number) =>
            valueToCheck === _value,
        lowOrEq: (valueToCheck: number, _value: number) =>
            valueToCheck <= _value,
        greatOrEq: (valueToCheck: number, _value: number) =>
            valueToCheck >= _value,
    };
    return !!conditions[condition as keyof typeof conditions]
        ? conditions[condition as keyof typeof conditions](value, valueCompare)
        : false;
};

export const arrUpTo = (numberFrom: number, numberTo: number) => {
    if (Number.isSafeInteger(numberFrom) && Number.isSafeInteger(numberTo)) {
        return Array.from(Array(numberTo), (...params) => params[1] + numberFrom)
    }
    return []
}
