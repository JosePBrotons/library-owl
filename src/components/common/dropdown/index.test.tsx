import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from '.';
import { arrUpTo } from '../../../utils';

test('renders dropdown component correctly', () => {
    const props = {
        options: arrUpTo(18, 82),
        optionsDescription: 'Years',
        label: 'Age',
        selectedValue: '18 years',
        fieldName: 'age',
        select: jest.fn,
    };
    const dropdownCmp = renderer.create(<Dropdown {...props} />).toJSON();
    expect(dropdownCmp).toMatchSnapshot();
});
