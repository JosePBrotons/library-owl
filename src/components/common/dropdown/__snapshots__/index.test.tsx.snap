// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dropdown component correctly 1`] = `
<View
  forwardedRef={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(248, 248, 248, 1)",
      "borderRadius": 5,
      "color": "rgba(69, 69, 69, 1)",
      "flexDirection": "row",
      "height": 40,
      "marginBottom": 10,
      "paddingHorizontal": 10,
      "width": "80%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(57, 172, 233, 1)",
          "fontFamily": "Nunito-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "paddingLeft": 8,
        }
      }
    >
      18 years
    </Text>
  </View>
</View>
`;
